packages/create-plugin/src/utils/utils.templates.ts

```diff
@@ -88,23 +92,70 @@ export function renderTemplateFromFile(templateFile: string, data?: any) {
   return renderHandlebarsTemplate(fs.readFileSync(templateFile).toString(), data);
 }
 
-export function getTemplateData(): TemplateData {
+export function buildProxyFromJson() {
   const pluginJson = getPluginJson();
+  return new Proxy(pluginJson, {
+    get(target, prop: string | symbol) {
+      if (prop === 'pluginId') {
+        return target.id;
+      } else if (prop === 'pluginName') {
+        return target.name;
+      } else if (prop === 'pluginDescription') {
+        return target.info?.description;
+      } else if (prop === 'hasBackend') {
+        return target.backend;
+      } else if (prop === 'hasGithubWorkflows') {
+        return undefined;
+      } else if (prop === 'hasGithubLevitateWorkflow') {
+        return undefined;
+      } else if (prop === 'orgName') {
+        return target.info?.author?.name;
+      } else if (prop === 'pluginType') {
+        return target.type;
+      } else if (prop === 'packageManager') {
+        return getPackageManagerWithFallback();
+      } else {
+        throw new Error(`Property ${String(prop)} not found in package.json.`);
+      }
+    },
+  });
+}
+export function buildProxyFromUserPrompt(target: CliArgs) {
+  return new Proxy(target, {
+    get(target: CliArgs, prop: string | symbol) {
+      if (prop in target) {
+        return target[prop as keyof CliArgs];
+      } else if (prop === 'pluginId') {
+        return normalizeId(target.pluginName, target.orgName, target.pluginType);
+      } else if (prop === 'packageManager') {
+        return getPackageManagerFromUserAgent();
+      } else if (prop === 'hasBackend') {
+        return target.hasOwnProperty('hasBackend') && target.hasBackend;
+      } else {
+        throw new Error(`Property ${String(prop)} not found in user's prompt.`);
+      }
+    },
+  });
+}
+
+export function getTemplateData(proxy: any): TemplateData {
```

Comment:
I think it would be nice to also type the input parameter of `getTemplateData()` (instead of using `any`). I think what would be the most straightforward is if the function would receive a typed object that explicitly defines what input information we need to generate the template data.